lib/grape/validations/validate_service.rb

```diff
@@ -0,0 +1,51 @@
+module Grape
+  module Validations
+    class ValidateService
```

Comment:
Yes, ValidationService named inappropriate. But it is not Validator. It does not assume responsibility, specific validation performed by specific Validator. ValidateService class purpose is move the relevant code of verify from Endpoint to Validations module. 

Another idea is to assume that responsibility by the Validations module object.

example：
1. lib/grape/validations.rb
```ruby
module Grape
  # Registry to store and locate known Validators.
  module Validations
    class << self
      attr_accessor :validators

      def validate!(endpoint, validators, params)
        errors = []
        validators.each do |validator|
          begin
            validator.validate!(params)
          rescue Grape::Exceptions::Validation => e
            errors << e
          end
        end
        fail Grape::Exceptions::ValidationErrors, errors: errors, headers: endpoint.header if errors.any?

        clean_params_index(params)
      end

      def clean_params_index(params)
        case params
        when Array
          params = params.map do |item|
            clean_params_index(item)
          end
        when Hash
          params.delete("_param_index")
          params.each do |key, value|
            params[key] = clean_params_index(value)
          end
        end
        params
      end
    end

    self.validators = {}

    # Register a new validator, so it can be used to validate parameters.
    # @param short_name [String] all lower-case, no spaces
    # @param klass [Class] the validator class. Should inherit from
    #   Validations::Base.
    def self.register_validator(short_name, klass)
      validators[short_name] = klass
    end

  end
end
```

2. lib/grape/endpoint.rb
```ruby
module Grape
  class Endpoint
    ...

    def run_validators(validators, params)
      @params = Validations.validate!(self, validators, params)
    end

    ...
  end
end
```